indico/modules/events/agreements/notifications.py

```diff
@@ -24,23 +24,23 @@ def make_email_template(template, agreement, email_body=None):
 
 @email_sender
 def notify_agreement_new(agreement, email_body=None, cc_addresses=None, from_address=None):
-    with agreement.event.force_event_locale():
+    with agreement.event.force_event_locale(user=agreement.user):
```

Comment:
I'd pass it as a posarg, not as a kwarg. It's just one argument and it seems to be pretty clear what it is anyway.